### ISSUE TITLE
[Feature Request] Monitoring learning rates

### ISSUE THREAD

**Issue by alexander-g** [poets-ai/elegy#124]
Elegy uses `optax` which is really nice but I am really missing a way to monitor the learning rate during training and maybe log it to tensorboard. 
Moreover, it's sometimes inconvenient to compute the right parameters for optax schedules because they use steps and not epochs. For example when I want the learning rate to drop at epoch 10, I first need to convert it to the correct step from the size of the dataset and the batch size.

I don't know how to solve the first issue but for the second one it would be nice to extend the `elegy.Model` API with parameters like `per_epoch_schedule=` and `per_step_schedule=` to which I would pass the optax schedules which would be updated at the end of the epoch or step correspondingly. Even better if I could specify something like `per_step_schedule="cosine_decay"` or `per_epoch_schedule=['0.1@50%', '0.01@90%']` which would automatically compute the correct step or epoch for me.

**Comment by cgarciae** [comment]
@alexander-g I've been thinking about this, I think optax's composable optimizers are a nice innovation but it makes monitoring tasks harder. A possible solution would be to leverage our internal [Optimizer](https://github.com/poets-ai/elegy/blob/3494cc7d495198f4c383d3560ea05df65bb669ff/elegy/model/model_base.py#L543) class and expose it + add some functionality to it, specifically, add a way to define the learning rate schedule, with this in place it could be automatically added to the metrics or we could create a `Callback` to do this.

Some details:
* Our `Optimizer` is a Module that wraps an `optax` optimizer and manages its state.
* We could have `Optimizer` keep track of a `step` variable and optionally calculate `epoch` if `steps_per_epoch` is available.
* We could have `Optimizer` accepts a `learning_rate` function of the form:
  ```python
  def learning_rate(step: int, epoch: Optional[int]) -> float:
  ```
  to define the schedule.
* Optimizer could have a `lr()` method which yields the current learning rate, either a `Callback` or the `Model` class could use this to add it to the `logs`.

What do you think?

**Comment by alexander-g** [comment]
Yes sure, looks nice.
I'll take a look if I can implement it.

**Comment by cgarciae** [comment]
Closed by #135

**Comment by cgarciae** [comment]
3
